### PR TITLE
Add ca.vlacroix.Tally

### PR DESCRIPTION
**Pull request by vtrlx**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
  **This is a GNOME application which gives the user an arbitrary number of tally counters.**

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: ca.vlacroix.Tally.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		"buildsystem": "simple",
+		"build-commands": [
+			"make",
+			"make install INSTALL_TOP=/app"
+		],
+		"sources": [{
+			"type": "archive",
+			"url": "https://lua.org/ftp/lua-5.4.7.tar.gz",
+			"sha256": "9fbf5e28ef86c69858f6d3d34eccc32e911c1a28b4120ff3e84aaa70cfbf1e30"
+		}]
```
Comment (bbhtt) [inline ca.vlacroix.Tally.json:24]: is it possible to use this through the shared modules ? https://github.com/flathub/shared-modules/tree/master/lua5.4

I can bump the version if you need 5.4.7 specifically, it has 5.4.6

File: ca.vlacroix.Tally.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	}, {
+		"name": "lgi",
+		"buildsystem": "simple",
+		"build-commands": [
+			"make -C lgi LUA_VERSION=5.4",
+			"make -C lgi install LUA_VERSION=5.4 PREFIX=/app"
+		],
```
Comment (bbhtt) [inline ca.vlacroix.Tally.json:31]: Use `no-autogen: true`, `make-args` and `make-install-args` and drop `buildsystem: simple`.

File: ca.vlacroix.Tally.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+			"url": "https://github.com/lgi-devs/lgi/archive/e06ad94c8a1c84e3cdb80cee293450a280dfcbc7.zip",
+			"sha256": "003984a7a33236cc2368bf8c87905b6867d1292f844df236051a7b45f134d272"
+		}]
+	}, {
+		"name": "tally",
+		"buildsystem": "simple",
+		"build-commands": [
+			"make",
+			"make install"
+		],
```
Comment (bbhtt) [inline ca.vlacroix.Tally.json:43]: Same as above.

File: ca.vlacroix.Tally.json (hunk 2 above)
Comment (vtrlx) [reply ca.vlacroix.Tally.json]: TIL. Thank you!

File: ca.vlacroix.Tally.json (hunk 1 above)
Comment (vtrlx) [reply ca.vlacroix.Tally.json]: I did not specifically need 5.4.7, that was just the latest release and so it's what I chose when I first started building this with Flatpak. I'm quite new to using Flatpak and so I was unfamiliar with these concepts until you suggested them. I've now made the fixes you suggested.

**Comment by bbhtt** [comment]
bot, build ca.vlacroix.Tally

You can start the build as you fix issues.

File: ca.vlacroix.Tally.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+				"type": "archive",
+				"url": "https://github.com/lgi-devs/lgi/archive/e06ad94c8a1c84e3cdb80cee293450a280dfcbc7.zip",
+				"sha256": "003984a7a33236cc2368bf8c87905b6867d1292f844df236051a7b45f134d272"
+			}]
+		},
+		{
+			"name": "tally",
+			"no-autogen": true,
+			"make-args": [],
+			"make-install-args": [],
```
Comment (bbhtt) [inline ca.vlacroix.Tally.json:30]:
```suggestion
			"make-args": [],
			"make-install-args": [],
```

You can remove these two if it works with the default.

File: ca.vlacroix.Tally.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+		"--share=ipc",
+		"--socket=fallback-x11",
+		"--socket=wayland"
+	],
+	"modules": [
+		"shared-modules/lua5.4/lua-5.4.json",
+		{
+			"name": "lgi",
+			"no-autogen": true,
+			"make-args": ["-C", "lgi", "LUA_VERSION=5.4"],
```
Comment (bbhtt) [inline ca.vlacroix.Tally.json:18]: is the makefile in `lgi` directory ? Then you need `subdir: lgi`

Comment (vtrlx) [reply ca.vlacroix.Tally.json]: Helpful again. Thank you!

**Comment by bbhtt** [comment]
bot, build ca.vlacroix.Tally

**Pull request closed without merging**